crates/block-producer/src/block_builder/mod.rs

```diff
@@ -143,3 +148,31 @@ where
         Ok(())
     }
 }
+
+struct BlockProducer<BB> {
+    pub mempool: Arc<Mutex<Mempool>>,
+    pub block_interval: tokio::time::Duration,
+    pub block_builder: BB,
+}
+
+impl<BB: BlockBuilder> BlockProducer<BB> {
+    pub async fn run(self) {
+        let mut interval = tokio::time::interval(self.block_interval);
+        interval.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Delay);
```

Comment:
Should the delay behavior here not be `Burst`? I'm thinking, if building a block took longer than expected, we should try to build the next block as soon as possible rather than wait until the next multiple of the interval.

Reply:
It depends on how we want to handle missed blocks.

`Burst` if we believe this is a temporary overload and want to catch up the missed blocks. This will result in the most amount of non-periodic blocks, but once caught up we will be back on the correct average block time.

`Skip` if we want the timing to be perfect, but are okay with skipping a block every now and then.

`Delay` if we are okay with a single imperfect block time, and the subsequent blocks should realign to the time. `Delay` here just means that the interval should realign with the last tick. So if we had block time `2` then good blocks `[0,2,4,6]` and a slow block at `9` then the next blocks would be out at `[11, 13, 15]` instead of trying to squeeze a quicker block at `10` (`burst`).

This is less important for missing a single block, but in the case where we miss multiple blocks then things are a bit weird I think.

Reply:
Follow-up task in #519